### PR TITLE
Inv 40 main page

### PR DESCRIPTION
**Pull request by nskoretz**
## Changes
- Fix login page
- Commented out a route for /signup that was causing errors
- Added route to /home and Home.vue
- Created Home.vue that is the Vue version of "main.html"
- Edited table style for title alignment

image

## Notes
- Right now the items in the table are hardcoded. Waiting for the API to catch up so that we can send a request to /getSales or something on pageload and retrieve a list of items to display in the table.
- This page is navigable on the Node.js dev server at the url http://localhost:8080/#/home .

### CODE REVIEW

**Comment by HugoKlepsch** [comment]
Screen Shot 2019-07-15 at 3 11 17 PM
Nice quick review

File: server/package-lock.json [hunk 1]
```diff
@@ -19,20 +19,20 @@
...
-        "@babel/traverse": "7.5.0",
-        "@babel/types": "7.5.0",
-        "convert-source-map": "1.6.0",
-        "debug": "4.1.1",
-        "json5": "2.1.0",
-        "lodash": "4.17.14",
-        "resolve": "1.11.1",
-        "semver": "5.7.0",
-        "source-map": "0.5.7"
+        "@babel/code-frame": "^7.0.0",
```
Comment (HugoKlepsch) [inline server/package-lock.json:22]: I wonder why it keeps adding and removing the ^ at the front. 🤔

File: server/src/views/Home.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        <strong>Sales this Month</strong>
+        <table>
+          <tr>
+            <th v-for="header in salesTable.headers">{{ header }}</th>
+          </tr>
+          <template v-for="itemObj in salesTable.defaultItems">
+            <tr>
+              <td>{{ itemObj.item }}</td>
+              <td>{{ itemObj.buyPrice }}</td>
+              <td>{{ itemObj.sellPrice }}</td>
```
Comment (HugoKlepsch) [inline server/src/views/Home.vue:20]: I like this. Feels like Jinja templating server side.

File: server/src/views/Home.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        headers: [ 'Item', 'Buy Price', 'Sell Price' ],
+        defaultItems: [
+          { item: 'Thing1', buyPrice: '$100', sellPrice: '$120' },
+          { item: 'Anotha One', buyPrice: '$6.99', sellPrice: '$101.50' },
+        ],
+      },
+    };
+  }, // }}}
+})
+export default class Home extends Vue {}
```
Comment (HugoKlepsch) [inline server/src/views/Home.vue:56]: I wonder what you can put here. 🤔

File: server/src/views/Login.vue [hunk 4]
```diff
@@ -6,27 +6,41 @@
...
-              <input id='username' type='text' name='username' class='form-control' placeholder='Username'
-                     required pattern='[a-zA-Z0-9_\-]{4,30}' autofocus>
-            </div>
-            <div class='form-group'>
-              <input id='password' type='password' name='password' class='form-control' placeholder='Password'
-                     required pattern='.{4,100}'>
-            </div>
+        <div class='form'>
+          <input
+            autofocus
```
Comment (HugoKlepsch) [inline server/src/views/Login.vue:11]: This mans really alphabetizing the attributes. :+1:

File: server/src/views/Login.vue [hunk 5]
```diff
@@ -49,8 +63,27 @@ import Navbar from '@/components/Navbar.vue'; // @ is an alias to /src
...
   },
+  methods: {
+    attemptLogin( event ) {
+      console.log( event );
+      this.$http.post('/login', {
+        username: this.$data.loginData.username,
+        password: this.$data.loginData.password,
+      }).then((res) => {
+        console.log('help me');
+        console.log(res);
```
Comment (HugoKlepsch) [inline server/src/views/Login.vue:80]: Ideally we would route to the main page here, but this is fine for now.

Comment (HugoKlepsch) [reply server/src/views/Login.vue:80]: Also, why does this seem so easy when I'm not writing it? 😂

**Review by HugoKlepsch — APPROVED** [review]
This looks good. I'll see if I can use this as a pattern for the signup page.

**Comment by nskoretz** [comment]
@HugoKlepsch Lol, you know it's just package-lock acting up. :joy:

File: server/package-lock.json (hunk 1 above)
Comment (nskoretz) [reply server/package-lock.json:22]: _The world may never know_ . But seriously maybe I should do a quick Google, do you think there are some settings we can set that'll say, be more gentle with the package-lock? Hopefully

File: server/src/views/Home.vue (hunk 3 above)
Comment (nskoretz) [reply server/src/views/Home.vue:56]: In one of the TS examples they did but some declarations and stuff there. Something more to learn.

File: server/src/views/Login.vue (hunk 5 above)
Comment (nskoretz) [reply server/src/views/Login.vue:80]: We're we getting a 200 OK from that request? I guess only when Flask is running. But yeah, it'll redirect.

Comment (nskoretz) [reply server/src/views/Login.vue:80]: Honestly idk, maybe we were burnt out on Sun. Even the TS (shocked face) seems to be getting easier to read and write.

**Pull request merged**